### PR TITLE
Added some math funcs implementation

### LINKED ISSUE #508: Replace the exclusive term
1. In [src/QirRuntime/test/QIR-static/qir-driver.cpp](src/QirRuntime/test/QIR-static/qir-driver.cpp) see comment for line 60 (`bool dummy);`).
2. See also [this](https://github.com/microsoft/qsharp-runtime/pull/500#pullrequestreview-586077737).

1 
> In general, style guide is to avoid the term "dummy," as that can be an exclusive term; something like "placeholder" or even "_" is strongly preferred.
> Our own style guide on this isn't public, unfortunately, but it's worth noting both that our code of conduct explicitly bans discriminatory language, and that the term "dummy" is widely considered to fall under that definition. In particular, both Google and Twitter's style guides exclude "dummy" explicitly.
To be clear, I don't object to the functionality of having a placeholder argument like this, but I would strongly suggest that name be replaced as soon as is reasonably possible.

2 "Ensure exclusive language is not used in codebase"

### PR DESCRIPTION
**Pull request by kuzminrobin**
The first step for #501 
Resolves #508

### CODE REVIEW

File: src/QirRuntime/lib/QIR/bridge-qis.ll
Comment (irinayat-MS) [inline src/QirRuntime/lib/QIR/bridge-qis.ll:285]: As we discussed, it might make sense have a separate .ll file for math functions, but we can look into this later as it's basically a build structure refactor.

**Review by cgranade — COMMENTED** [review]
This looks reasonable to me, but I'm not an LLVM expert per se so I'll leave it to others to approve. In any case, it looks like the tests are all at the C level, but it would be good to have integration tests similar to https://github.com/microsoft/qsharp-runtime/blob/0cf4a5e168e0ab9b9d6ecdb176e25548c67bea7f/src/QirRuntime/test/QIR-dynamic/qir-test-random-lnx.ll#L90 that ensure that the new intrinsic works correctly when called from QIR files generated by the Q# compiler.

File: src/QirRuntime/test/QIR-static/qir-test-math.cpp
Comment (irinayat-MS) [inline src/QirRuntime/test/QIR-static/qir-test-math.cpp:5]: I'm not sure I like this direct testing of the bridge. While it has some value, I'd prefer to add a qs file that we'd use to generate QIR and then test that. This way we'll also check that our expectation of the generated signatures matches the compiler's output.

File: src/QirRuntime/lib/QIR/bridge-qis.ll
Comment (kuzminrobin) [reply src/QirRuntime/lib/QIR/bridge-qis.ll:285]: #505
(for now I would like to postpone that refactoring/maintenance)

File: src/QirRuntime/test/QIR-static/qir-test-qubits-results.qs
Comment (kuzminrobin) [inline src/QirRuntime/test/QIR-static/qir-test-qubits-results.qs:1]: Clarification: This change enables the (C/C++) syntax highlighting in my editor for this .qs file.
Let me know if you would like me to remove this change.

**Comment by kuzminrobin** [comment]
> This looks reasonable to me, but I'm not an LLVM expert per se so I'll leave it to others to approve. In any case, it looks like the tests are all at the C level, but it would be good to have integration tests similar to
> 
> https://github.com/microsoft/qsharp-runtime/blob/0cf4a5e168e0ab9b9d6ecdb176e25548c67bea7f/src/QirRuntime/test/QIR-dynamic/qir-test-random-lnx.ll#L90
> 
> that ensure that the new intrinsic works correctly when called from QIR files generated by the Q# compiler.

Moved the tests to .qs file.

File: src/QirRuntime/test/QIR-static/qir-test-math.cpp
Comment (kuzminrobin) [reply src/QirRuntime/test/QIR-static/qir-test-math.cpp]: Moved the test to the .qs file.

File: src/QirRuntime/test/QIR-static/qir-test-arrays.qs
Comment (kuzminrobin) [inline src/QirRuntime/test/QIR-static/qir-test-arrays.qs:38]: FYI: For me the compiler didn't crash.

File: src/QirRuntime/lib/QIR/intrinsicsMath.cpp
Comment (irinayat-MS) [inline src/QirRuntime/lib/QIR/intrinsicsMath.cpp:10]: The declarations are unnecessary here. Maybe, extract into a header file so
a) it's easier to find them (people often start exploration of a codebase from the headers)
b) when/if more complicated implementations are added, the header can be included into the files that test them

File: src/QirRuntime/test/QIR-dynamic/CMakeLists.txt
Comment (irinayat-MS) [inline src/QirRuntime/test/QIR-dynamic/CMakeLists.txt:39]: The idea of the "dynamic" tests was to validate that can link against the dynamic lib (qdk) without any knowledge about the static libs that go into it. If it's failing to link after your changes, it means that we aren't building the qdk component correctly. Ideally, should fix the build. Or log a bug and disable the component for now.

File: src/QirRuntime/test/QIR-static/CMakeLists.txt
Comment (irinayat-MS) [inline src/QirRuntime/test/QIR-static/CMakeLists.txt:24]: Nit: maybe, move the closing parenthesis to a new line, so future additions look cleaner in diffs?

File: src/QirRuntime/test/QIR-static/compiler/QirTarget.qs
Comment (irinayat-MS) [inline src/QirRuntime/test/QIR-static/compiler/QirTarget.qs:8]: Should these be marked `@Inline`? (Not a big deal, probably, because eventually we want to use compiler's target-specific qs files anyway.)

File: src/QirRuntime/test/QIR-static/qir-driver.cpp
Comment (irinayat-MS) [inline src/QirRuntime/test/QIR-static/qir-driver.cpp:61]: Will catch's filters still work with category specified like this?

File: src/QirRuntime/test/QIR-static/qir-test-arrays.qs
Comment (irinayat-MS) [reply src/QirRuntime/test/QIR-static/qir-test-arrays.qs:38]: You are probably on a newer version of the compiler, which means whatever bug it was, it's fixed. Good :)

File: src/QirRuntime/test/QIR-static/qir-test-math.qs
Comment (irinayat-MS) [inline src/QirRuntime/test/QIR-static/qir-test-math.qs:12]: You shouldn't need nested `ifs` here because you are not allocating qubits, early returns should compile just fine:

```
if (2.0 != Sqrt(  4.0) )    { return 1; }
if (3.0 != Sqrt(  9.0) )    { return 2; }
// etc.
```

File: src/QirRuntime/test/QIR-static/qir-test-qubits-results.qs
Comment (irinayat-MS) [reply src/QirRuntime/test/QIR-static/qir-test-qubits-results.qs:1]: Would you mind adding the licensing boilerplate instead (all our source files are required to have it, but apparently I've missed this one)?

```
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
```

Comment (cgranade) [reply src/QirRuntime/test/QIR-static/qir-test-qubits-results.qs:1]: As a slight nit, the most up to date version of the header does not have "All rights reserved." It's not essential to go fix in existing headers, but is recommended for all new headers.

File: src/QirRuntime/test/QIR-static/qir-test-math.qs
Comment (cgranade) [inline src/QirRuntime/test/QIR-static/qir-test-math.qs:9]: Q# style is to have braces in the same line, as opposed to C#'s next-line style. I'd also suggest that the name "TestSqrt" sounds like a verb, whereas Q# style is for functions to be named as nouns.

File: src/QirRuntime/test/QIR-static/qir-test-math.qs
Comment (cgranade) [reply src/QirRuntime/test/QIR-static/qir-test-math.qs:12]: That would also have the beneficial effect of eliminating the `mutable` variable.

File: src/QirRuntime/test/QIR-static/qir-driver.cpp
Comment (cgranade) [inline src/QirRuntime/test/QIR-static/qir-driver.cpp:61]: In general, style guide is to avoid the term "dummy," as that can be an exclusive term; something like "placeholder" or even "_" is strongly preferred.

File: src/QirRuntime/test/QIR-static/qir-test-arrays.qs
Comment (cgranade) [inline src/QirRuntime/test/QIR-static/qir-test-arrays.qs:36]: Similarly here.

File: src/QirRuntime/lib/QIR/bridge-qis.ll
Comment (guenp) [inline src/QirRuntime/lib/QIR/bridge-qis.ll:290]: Can we drop this from the PR or is this commented to be added back in later? If so, would you mind adding a note on that with optionally a link to a GH ticket?

File: src/QirRuntime/lib/QIR/bridge-qis.ll
Comment (guenp) [inline src/QirRuntime/lib/QIR/bridge-qis.ll:323]: Same comment here; can we drop this and if not, could we add a note and GH issue #?

**Review by guenp — COMMENTED** [review]
Just some comments; thanks for implementing this!

File: src/QirRuntime/test/QIR-static/qir-driver.cpp
Comment (kuzminrobin) [reply src/QirRuntime/test/QIR-static/qir-driver.cpp:61]: For me it does work like this.
```
<test.exe> 
<test.exe> [qir]
<test.exe> [qir.arr1d]
```
And I had a test failure that was showing up on exactly one testcase (coincidentally exactly this one ;-) ). So I wanted to fix-and-run multiple times one particular testcase. So I added this finer granularity.

File: src/QirRuntime/test/QIR-dynamic/CMakeLists.txt
Comment (irinayat-MS) [reply src/QirRuntime/test/QIR-dynamic/CMakeLists.txt:39]: as discussed offline: add the export decls on the new APIs

File: src/QirRuntime/lib/QIR/bridge-qis.ll
Comment (kuzminrobin) [reply src/QirRuntime/lib/QIR/bridge-qis.ll:290]: This is work in progress to be added in the next commit/PR.
(I was working on it when I got the first feedback about Sqrt, so I postponed Ceil until I resolve all for Sqrt)

File: src/QirRuntime/lib/QIR/bridge-qis.ll
Comment (kuzminrobin) [reply src/QirRuntime/lib/QIR/bridge-qis.ll:323]: This is work in progress to be added in the next commit/PR.
(I was working on it when I got the first feedback about Sqrt, so I postponed Ceil until I resolve all for Sqrt)

File: src/QirRuntime/test/QIR-static/qir-driver.cpp
Comment (irinayat-MS) [reply src/QirRuntime/test/QIR-static/qir-driver.cpp:61]: FYI, to run just a single test, you can use its description string as the filter

File: src/QirRuntime/lib/QIR/bridge-qis.ll
Comment (cgranade) [reply src/QirRuntime/lib/QIR/bridge-qis.ll:323]: Ceiling shouldn't be needed at all post-#497, though? Ceiling, Floor, and Round are all expressed in terms of Truncate, such that that's the only intrinsic needed to support those.

File: src/QirRuntime/test/QIR-static/compiler/QirTarget.qs
Comment (cgranade) [inline src/QirRuntime/test/QIR-static/compiler/QirTarget.qs:12]: Are these intended to be new functions in the public Q# API? If so, they will need to go through API review first.

File: src/QirRuntime/test/QIR-static/qir-test-qubits-results.qs
Comment (kuzminrobin) [reply src/QirRuntime/test/QIR-static/qir-test-qubits-results.qs:1]: Fixed.

File: src/QirRuntime/lib/QIR/intrinsicsMath.cpp
Comment (cgranade) [inline src/QirRuntime/lib/QIR/intrinsicsMath.cpp:1]:
```suggestion
// Copyright (c) Microsoft Corporation.
```

File: src/QirRuntime/test/QIR-static/qir-driver.cpp
Comment (kuzminrobin) [reply src/QirRuntime/test/QIR-static/qir-driver.cpp:61]: I would prefer to refrain dealing with this case of "dummy" in this PR since "dummy" is out of this PR's scope. This case of "dummy" has important functionality behind it (not knowing which can result in a loss of a few hours for a new-comer), and if I touch this "dummy" I feel obliged to document in details that important functionality.

However I'm inclined to agree about the use of "dummy" in general, and would like to know more about it and other similar identifiers/names. Any direct link about more info? (preferably reachable from [Coding Style](https://ms-quantum.visualstudio.com/Quantum%20Program/_wiki/wikis/Quantum-OKRs.wiki/487/Coding-Style))

File: src/QirRuntime/test/QIR-static/qir-test-arrays.qs
Comment (kuzminrobin) [reply src/QirRuntime/test/QIR-static/qir-test-arrays.qs:36]: I will focus on _my_ changes for now.
However I will keep your code style recommendations in mind for the lines/functions/operations I touch. Thank you for sharing your knowledge. Please keep sharing, I appreciate that.

File: src/QirRuntime/lib/QIR/bridge-qis.ll
Comment (kuzminrobin) [reply src/QirRuntime/lib/QIR/bridge-qis.ll:323]: Thanks for letting me know on time, Chris. Removing Ceil from my plate, adding Truncate.

File: src/QirRuntime/test/QIR-static/compiler/QirTarget.qs
Comment (kuzminrobin) [reply src/QirRuntime/test/QIR-static/compiler/QirTarget.qs:12]: (FYI: @bettinaheim, @irinayat-MS, @guenp)
Bettina,
What do I need to do to get those through API review?

I assume that Q#'s `Sqrt()` (inspired by C#'s [Math.Sqrt](https://docs.microsoft.com/en-us/dotnet/api/system.math.sqrt?view=net-5.0)) is expected to work with `NaN` and `PositiveInfinity` also (i.e. the Q#'s `Sqrt()` requires them in Q#). Thus it seems to me reasonable for the functionality of NaN, PositiveInfinity/Infinity, `IsNan()`, `IsInf()` to _become new functions in the public Q# API_ (in simulators can be implemented through the following C++'s <cmath> calls/constants - [`NAN`/`nan()`/`quiet_NaN()`](https://en.cppreference.com/w/cpp/numeric/math/NAN), [`INFINITY`/`infinity`](https://en.cppreference.com/w/cpp/numeric/math/INFINITY), [`isnan()`](https://en.cppreference.com/w/cpp/numeric/math/isnan), [`isinf()`](https://en.cppreference.com/w/cpp/numeric/math/isinf)).

Chris, thanks!

File: src/QirRuntime/lib/QIR/intrinsicsMath.cpp
Comment (kuzminrobin) [reply src/QirRuntime/lib/QIR/intrinsicsMath.cpp:1]: Done (including for my files added to tests).
Where can I get more info about this? (to know in the future what is recommended or not recommended)

File: src/QirRuntime/lib/QIR/intrinsicsMath.cpp
Comment (kuzminrobin) [reply src/QirRuntime/lib/QIR/intrinsicsMath.cpp:10]: Fixed.

File: src/QirRuntime/test/QIR-dynamic/CMakeLists.txt
Comment (kuzminrobin) [reply src/QirRuntime/test/QIR-dynamic/CMakeLists.txt:39]: Fixed.

File: src/QirRuntime/test/QIR-static/CMakeLists.txt
Comment (kuzminrobin) [reply src/QirRuntime/test/QIR-static/CMakeLists.txt:24]: Fixed.

File: src/QirRuntime/test/QIR-static/compiler/QirTarget.qs
Comment (kuzminrobin) [reply src/QirRuntime/test/QIR-static/compiler/QirTarget.qs:8]: Fixed.

File: src/QirRuntime/test/QIR-static/qir-test-math.qs
Comment (kuzminrobin) [reply src/QirRuntime/test/QIR-static/qir-test-math.qs:12]: Fixed.

File: src/QirRuntime/test/QIR-static/qir-test-math.qs
Comment (kuzminrobin) [reply src/QirRuntime/test/QIR-static/qir-test-math.qs:9]: Fixed.

File: src/QirRuntime/test/QIR-static/compiler/QirTarget.qs
Comment (cgranade) [reply src/QirRuntime/test/QIR-static/compiler/QirTarget.qs:14]: Please see the [Q# API Design Process](https://github.com/microsoft/QuantumLibraries/wiki/Q%23-API-Design-Process) documentation on the Q# libraries wiki for more details, but roughly the Q# API Review is a monthly meeting that I run with the Q# API review board (currently @guenp, @msoeken, @bettinaheim, and @efratshabtai) to make sure that all new or modified public Q# APIs meet our [style guide](https://docs.microsoft.com/quantum/resources/contributing/style-guide) and our [API design principles](https://docs.microsoft.com/azure/quantum/contributing-api-design-principles).

Following that process, if we want to add these functions to the public Q# API, my suggestion would be to open a proposal on the [microsoft/quantumlibraries repo](https://github.com/microsoft/QuantumLibraries), so that it can be scheduled for inclusion in the March review. Should the Q# API review approve the new API, and should @bettinaheim sign off on the corresponding impact on the runtime interface, we can then implement those new functions as `body intrinsic` callables in the QSharpFoundation project on the qsharp-runtime repo, with the changes here in QirTarget.qs and bridge-qis.ll being used to implement those new functions.

Comment (kuzminrobin) [reply src/QirRuntime/test/QIR-static/compiler/QirTarget.qs:14]: See [413](https://github.com/microsoft/QuantumLibraries/issues/413) in QuantumLibraries.
I propose those for the Q# API Design Review.
@cgranade, @bettinaheim.

File: src/QirRuntime/lib/QIR/intrinsicsMath.cpp
Comment (cgranade) [inline src/QirRuntime/lib/QIR/intrinsicsMath.cpp:28]: As per your note below, does this introduce a discrepancy with existing semantics for Q#'s `ArcTan2` function?

File: src/QirRuntime/test/QIR-static/qir-driver.cpp
Comment (irinayat-MS) [reply src/QirRuntime/test/QIR-static/qir-driver.cpp:61]: What does the negative filter do? (test.exe ~[qir]) Does it still filter our the whole category?

**Review by irinayat-MS — APPROVED** [review]
Looks good to me on the QirRuntime's side of things. Signing off modulo Chris' comments on the semantics of arctan2.

File: src/QirRuntime/test/QIR-static/compiler/Constants.qs
Comment (kuzminrobin) [reply src/QirRuntime/test/QIR-static/compiler/Constants.qs:9]: Yes. That's how I interpreted the offline response.
I wrote:
> I want to add a test case to the .qs file in qsharp-runtime/src/QirRuntime/test/QIR-static and I want my test case to access the E() constant defined in Microsoft.Quantum.Math namespace in qsharp-runtime\src\Simulation\QSharpFoundation\Math\Constants.qs.
However, in order to generate the .ll files from .qs tests, the qsharp-runtime/src/QirRuntime/test/QIR-static/generate.py does not use libraries of the Q# compiler, it uses just a minimal set of a few .qs files (in qsharp-runtime/src/QirRuntime/test/QIR-static/compiler).
What is the best/right way to build the .qs tests that use the Q# libraries?

The reply was:
> Assuming you are asking for the purpose of testing the functions implemented in the qir runtime (opposed to setting up integration tests), you could simple compile [the file where it is declared](https://github.com/microsoft/qsharp-runtime/blob/main/src/Simulation/QSharpFoundation/Math/Constants.qs) along with the .qs file for your test.

File: src/QirRuntime/test/QIR-static/qir-test-math.qs
Comment (kuzminrobin) [reply src/QirRuntime/test/QIR-static/qir-test-math.qs:57]: Looks like the [C#'s (.NET's) semantics](https://docs.microsoft.com/en-us/dotnet/api/system.math.atan2?view=net-5.0) somewhat contradicts to itself:
On the one hand it says: 
> if x and y are either PositiveInfinity or NegativeInfinity, the method returns NaN

On the other hand it remarks:
> This method calls into the underlying C runtime, and the exact result or valid input range may differ between different operating systems or architectures.

The [C](https://en.cppreference.com/w/c/numeric/math/atan2) (and [C++](https://en.cppreference.com/w/cpp/numeric/math/atan2)) runtime returns NaN in one case only - "If either x is NaN or y is NaN".

For now I suspect that the phrase 
> if x and y are either PositiveInfinity or NegativeInfinity, the method returns NaN

is out-of-date even for C# (and for the whole .NET), and even in C# the behavior corresponds to the C (and C++) runtime.
So for now I would leave the implementation and documentation for Q# as is (since Guen is blocked by this missing functionality), and get back to the discussion later if needed.

I filed a documentation issue [77](https://github.com/microsoft/qsharp-language/issues/77) in qsharp-language.

File: src/QirRuntime/lib/QIR/intrinsicsMath.cpp
Comment (kuzminrobin) [reply src/QirRuntime/lib/QIR/intrinsicsMath.cpp:28]: See [this my reply](https://github.com/microsoft/qsharp-runtime/pull/500#discussion_r572467750).

File: src/QirRuntime/test/QIR-static/qir-test-math.qs
Comment (cgranade) [reply src/QirRuntime/test/QIR-static/qir-test-math.qs:57]: Confirming, it looks like the current behavior of Q#'s `ArcTan2` follows what I would expect from the cppreference docs that you shared:

image

This would be good to test for explicitly in the C#-based runtime, but given that the two implementations seem to agree, I agree that it makes sense to defer adding those tests until the next PR.

File: src/QirRuntime/test/QIR-static/qir-driver.cpp
Comment (kuzminrobin) [reply src/QirRuntime/test/QIR-static/qir-driver.cpp:61]: Yes, it filters our the whole [qir] category.

File: src/QirRuntime/test/QIR-static/qir-test-math.qs
Comment (kuzminrobin) [reply src/QirRuntime/test/QIR-static/qir-test-math.qs]: Fixed.

File: src/QirRuntime/test/QIR-static/qir-test-math.qs
Comment (kuzminrobin) [reply src/QirRuntime/test/QIR-static/qir-test-math.qs]: Yes. it indicates which test failed.
Added a comment.

Comment (kuzminrobin) [reply src/QirRuntime/test/QIR-static/qir-test-math.qs]: Fixed.

File: src/QirRuntime/test/QIR-static/qir-test-math.qs
Comment (kuzminrobin) [reply src/QirRuntime/test/QIR-static/qir-test-math.qs:57]: Thanks for confirming! Resolving this conversation for now (in this PR).

File: src/QirRuntime/test/QIR-static/qir-test-strings.qs
Comment (cgranade) [inline src/QirRuntime/test/QIR-static/qir-test-strings.qs:12]: You may be able to avoid mutability here, since `pauli` and `str` are each only used once:

```suggestion
        if "Pauli: PauliI" != $"Pauli: {PauliI}" { return 1; }     // The return value indicates which test case has failed.
```

**Review by cgranade — APPROVED** [review]
Approving to unblock @guenp, but we should as quickly as possible:

- [x] Ensure exclusive language is not used in codebase
- [ ] Extend QSharpFoundation unit tests to cover what you noticed about `ArcTan2`, so that we can prevent regressions that would cause QIR and C# runtimes to disagree

**Comment by kuzminrobin** [comment]
> Approving to unblock @guenp, but we should as quickly as possible:
> 
> * [x]  Ensure exclusive language is not used in codebase
> * [x]  Extend QSharpFoundation unit tests to cover what you noticed about `ArcTan2`, so that we can prevent regressions that would cause QIR and C# runtimes to disagree

Filed #508.
Added to #506:
> Extend QSharpFoundation unit tests to cover what has been noticed about ArcTan2, so that we can prevent regressions that would cause QIR and C# runtimes to disagree

File: src/QirRuntime/test/QIR-static/qir-test-strings.qs
Comment (kuzminrobin) [reply src/QirRuntime/test/QIR-static/qir-test-strings.qs]: Fixed.

File: src/QirRuntime/lib/QIR/bridge-qis.ll
Comment (irinayat-MS) [inline src/QirRuntime/lib/QIR/bridge-qis.ll:16]: Nit: Notice, that generated QIR doesn't actually use this type but uses i2 directly. My choice of not introducing type aliases was to match the generated code verbatim, in hope that it will be easier to find bugs like the wrong parameter in string creation. At some point I even had a script that attempted to automatically extract declared signatures from generated QIR and compare them to defined here but I couldn't get it reliable enough...

File: src/QirRuntime/lib/QIR/strings.cpp
Comment (irinayat-MS) [inline src/QirRuntime/lib/QIR/strings.cpp:141]: What does the QIR spec says for this? Shouldn't the call fail rather than return a random string?

Comment (kuzminrobin) [reply src/QirRuntime/lib/QIR/strings.cpp:141]: My impression is that the [QIR Spec](https://github.com/microsoft/qsharp-language/blob/main/Specifications/QIR/Data-Types.md) does not cover that case (I searched through all the occurrences of "pauli").
Thus I needed to invent something.

File: src/QirRuntime/lib/QIR/bridge-qis.ll
Comment (kuzminrobin) [reply src/QirRuntime/lib/QIR/bridge-qis.ll:16]: OK, I'll keep that in mind.

I'll admit that now the types `i2` and `%Pauli` are the same (`%Pauli` being an alias/typedef of `i2`). Before this change, the were, strictly speaking, _different types_ (`%Pauli` was a struct with one field of type `i2`), and casting to `i32` was causing compilation error.

File: src/QirRuntime/lib/QIR/strings.cpp
Comment (irinayat-MS) [reply src/QirRuntime/lib/QIR/strings.cpp:141]: Log a bug against the spec? Imho, it should cover error cases like this.

File: src/QirRuntime/lib/QIR/bridge-qis.ll
Comment (irinayat-MS) [reply src/QirRuntime/lib/QIR/bridge-qis.ll:16]: This is probably how %Pauli was defined in the earlier versions of generated QIR when I copied it over into the bridge. But because it was never actually used, the outdated type didn't come to my attention.

**Pull request merged**
